src/core/seat/input-manager.hpp

```diff
@@ -49,6 +49,12 @@ class input_manager_t
      */
     void configure_input_devices();
 
+    /**
+     * Calibrate a touch device with a matrix. This function does nothing
+     * if called with a device that is not a touch device.
+     */
+    void calibrate_touch_device(wlr_input_device *dev, std::string const & cal);
```

Comment:
This doesn't seem like it should be in input-manager ... Because this is device-type-specific (mapping inputs to outputs works for tablets, touch and pointer). Ideally speaking, we'd want to add an input-device class for touch devices and load their specific options there, just like we do for pointers: https://github.com/WayfireWM/wayfire/blob/master/src/core/seat/pointing-device.hpp

And then we can create this device-specific object here: https://github.com/WayfireWM/wayfire/blob/56532427bea34ed9d777708b52356220c366de73/src/core/seat/input-manager.cpp#L39

At the very least, it seems like this should be somewhere in the input device class.

Reply:
Hm, ok..